packages/@lwc/rollup-plugin/src/index.ts

```diff
@@ -0,0 +1,152 @@
+    stylesheetConfig?: any;
+}
+
+const IMPLICIT_DEFAULT_HTML_PATH = '@lwc/resources/empty_html.js';
+
+const EMPTY_IMPLICIT_HTML_CONTENT = 'export default void 0';
+const EMPTY_IMPLICIT_CSS_CONTENT = '';
+
+function isImplicitHTMLImport(importee: string, importer: string): boolean {
+    return (
+        path.extname(importer) === '.js' &&
+        path.extname(importee) === '.html' &&
+        path.dirname(importer) === path.dirname(importee) &&
+        path.basename(importer, '.js') === path.basename(importee, '.html')
+    );
+}
+
+function lwc(options: RollupLwcOptions = {}): Plugin {
+    let { rootDir, modules = [] } = options;
+    const { stylesheetConfig, sourcemap = false, environment = 'dom' } = options;
+
+    if (environment !== 'dom' && environment !== 'server') {
+        throw new TypeError(
+            `Invalid "environment" option. Received ${environment} but expected "dom", "server" or undefined.`
+        );
+    }
+
+    const filter = pluginUtils.createFilter(options.include, options.exclude);
+
+    return {
+        name: 'lwc',
+
+        buildStart({ input }) {
+            if (rootDir === undefined) {
+                if (Array.isArray(input)) {
+                    rootDir = path.dirname(input[0]);
+
+                    if (input.length > 1) {
+                        this.warn(
+                            `The "rootDir" option should be explicitly set when passing an "input" array to rollup. The "rootDir" option is implicitly resolved to ${rootDir}.`
+                        );
+                    }
+                } else {
+                    rootDir = path.dirname(Object.values(input)[0]);
+
+                    this.warn(
+                        `The "rootDir" option should be explicitly set when passing "input" object to rollup. The "rootDir" option is implicitly resolved to ${rootDir}.`
+                    );
+                }
+            } else {
+                rootDir = path.resolve(rootDir);
+            }
+
+            modules = [...modules, { dir: rootDir }, { npm: 'lwc' }];
+        },
+
+        resolveId(importee, importer) {
+            if (importer === undefined) {
+                return;
+            }
+
+            const isRelativeImport = importee.startsWith('.');
+            if (isRelativeImport) {
+                const importerExt = path.extname(importer);
+                const ext = path.extname(importee) || importerExt;
+
+                const normalizedPath = path.resolve(path.dirname(importer), importee);
+                const absPath = pluginUtils.addExtension(normalizedPath, ext);
+
+                if (isImplicitHTMLImport(normalizedPath, importer) && !fs.existsSync(absPath)) {
+                    return IMPLICIT_DEFAULT_HTML_PATH;
+                }
+
+                return absPath;
+            } else {
+                if (importee === 'lwc') {
+                    importee = environment === 'dom' ? '@lwc/engine-dom' : '@lwc/engine-server';
```

Comment:
how will this behave when you have a configuration that marks `lwc` as an external module?

Reply:
i made some experiments, and when a module is marked as `external` it does not go through `resolveId`, so this should be fine. pls confirm!

Reply:
Yes, you are right, if a module is marked as external, rollup doesn't even bother attempting to resolve it.